VFS/vfs.cpp

```diff
@@ -0,0 +1,192 @@
+                currnode->ext.ext_dir.files = NULL;
+                currnode = currnode->ext.ext_dir.files;
+                continue;
+            }
+            if (!(currnode->flags & FLAGS_ISDIR)) {
+                return false;
+            }
+            if (!(currnode->flags & FLAGS_LOADED)) {
+                vfs_load_contents(currnode,currpath);
+                continue;
+            }
+            bool found = false;
+            uint8_t nextslashin = 0;
+            for (size_t i = 0;true; i++)
+            {
+                if (path[currpathidx+i] == '/')
+                {
+                    nextslashin = i;
+                    break;
+                }else if(nextslashin == 0 && path[currpathidx+i] == 0){
+                    nextslashin = i;
+                    break;    
+                }
+            }
+        
+            for (size_t i = 0; i <currnode->ext.ext_dir.num_dirs;
+                 i++) {
+                    for (size_t i = 0;i < currnode->ext.ext_dir.num_dirs; i++) {
+                        if (strncmp(currnode->ext.ext_dir.files[i].name, path + (currpathidx + 1), nextslashin) == 0) {
+                            currpath[currpathidx] = '/';
+                            memcpy(&currpath[currpathidx + 1],currnode->ext.ext_dir.files[i].name, strlen(currnode->ext.ext_dir.files[i].name) + 1);
+                            currpathidx += strlen(currnode->ext.ext_dir.files[i].name) + 1;
+
+                            currnode = &currnode->ext.ext_dir.files[i];
+                            found = true;
+                            break;
+                        }
+                    }
+            }
+            if (!found) {
+                return false;
+            }
+        }
+    }else{
+        currnode->flags = FLAGS_PRESENT | FLAGS_ISDIR | FLAGS_LOADED;
+        currnode->name = "/";
+
+        currnode->ext.ext_dir.creation_date_unix_time = 0;
+        currnode->ext.ext_dir.last_modified_unix_time = 0;
+        currnode->ext.ext_dir.disk_id = 0;
+        currnode->ext.ext_dir.owner_group = 0;
+        currnode->ext.ext_dir.owner_user = 0;
+        currnode->ext.ext_dir.num_dirs = 1;
+
+        currnode->ext.ext_dir.files = NULL;
+    }
+    currnode->flags |= FLAGS_MOUNT_POINT;
+    struct dir_report rep = iterate_dir(disk_id,"/");
+    currnode->ext.ext_dir.num_dirs = rep.num_entries;
+    currnode->ext.ext_dir.files = rep.entries;
+    
+    return true;
+}
+struct file_buffer vfs_read(char* path,uint64_t offset,uint64_t size){
+    struct node* file = vfs_inspect(path);
+    return read_file(file->ext.ext_file.disk_id,(char*)file->ext.ext_file.contents_on_disk_path,offset,size);
+}
+void vfs_write(char* path,struct file_buffer buf,uint64_t offset){
+    struct node* file = vfs_inspect(path);
+    write_file(file->ext.ext_file.disk_id,(char*)file->ext.ext_file.contents_on_disk_path,offset,buf);
+}
+
+struct node *vfs_inspect(char *path) {
+    if (strcmp(path, "/") == 0) {
```

Comment:
`strcmp()` can be unsafe, using `strncmp()` is more desirable

Reply:
so you want `/bin` be root?????
that is stupid

Reply:
I thinki you dont understand what this check is for

Reply:
wrong repo btw

Reply:
> wrong repo btw

port it yourself then, my changes are here, YOU CAN DO IT